maintenance-filter-server/src/main/java/org/ccci/maintenance/util/MaintenanceWindowDao.java

```diff
@@ -0,0 +1,18 @@
+package org.ccci.maintenance.util;
+
+import org.ccci.maintenance.MaintenanceWindow;
+import org.joda.time.DateTime;
+
+/**
+ * @author Matt Drees
+ */
+public interface MaintenanceWindowDao
+{
+    MaintenanceWindow getActiveMaintenanceWindow(String filterName, DateTime currentDateTime);
+
+    /** retrieves the window identified by the given id, or null if it does not exist */
+    OwnedMaintenanceWindow getMaintenanceWindowById(String id);
```

Comment:
What are you thoughts on using `Optional<OwnedMaintenanceWindow` here?  I'm growing to like the Optional pattern more and more b/c it reduces null checks all throughout my code.  Maybe it's desirable here though b/c it would add a Guava dependency not only to this lib but to any calling lib as well.  I could go either way.. what do you think?